sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,283 @@ def reduce_inequalities(inequalities, symbols=[]):
+        if not rel.rhs.is_constant():
+            raise ValueError("Right hand side of relation must be constant")
+
+        free_sym = free_sym | rel.free_symbols
+
+        if type(rel) == Le:
+            A.append(rel.lhs)
+            B.append(rel.rhs)
+        elif type(rel) == Ge:
+            A.append(-rel.lhs)
+            B.append(-rel.rhs)
+        elif type(rel) == Eq:
+            # x = 3 can be represented as x <= 3 and x >= 3
+            A.append(rel.lhs)
+            B.append(rel.rhs)
+            A.append(-rel.lhs)
+            B.append(-rel.rhs)
+        else:
+            raise TypeError(f"Unrecognized relational: {rel}")
+
+    free_sym = sorted(free_sym, key=lambda v: str(v)) # order symbols lexicographically
+
+    A = Matrix([[row.coeff(sym) for sym in free_sym] for row in A])
+    B = Matrix(B)
+    C = Matrix([[objective.coeff(sym) for sym in free_sym]])
+
+    return A, B, C
+
+# Maximize Cx + D constrained with Ax <= B and x >= 0
+# Minimize y^{T}B constrained with y^{T}A >= C^{T} and y >= 0
+def linear_programming(constraints, objective, random_seed=0):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method.
+
+    Explanation
+    ===========
+
+    When x is a column vector of variables, y is a column vector of dual variables,
+    and when the objective is either:
+
+    *) Maximizing Cx constrained to Ax <= B and x >= 0.
+    *) Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
+
+    The method returns a triplet of solutions optimum, argmax and argmax_dual where
+    optimum is the optimum solution, and argmax/ argmax_dual give the
+    values for x and y, respectively.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities
+        The lhs of these inequalities are equivalent to the matrix A and the rhs
+        are equivalent to the column vector B.
+
+    objective : a linear function to maximize
+        The objective function is represented by the row vector C.
+
+    random_seed : int
+        Determines the random pivoting behavior of the simplex method
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linear_programming
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual = linear_programming([r1,r2,r3], x+y+5*z)
+    >>> optimum
+    11/3
+    >>> argmax
+    [0, 1/3, 2/3]
+    >>> argmax_dual
+    [0, 2/3, 1]
+
+    The method also works with non-integer coefficients (like sqrt(2)).
+
+    >>> from sympy import sqrt
+    >>> r1 = y + sqrt(2)*z <= sqrt(3)
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual = linear_programming([r1,r2,r3], x+y+sqrt(5)*z)
+    >>> optimum
+    3
+    >>> argmax
+    [2, 1, 0]
+    >>> argmax_dual
+    [0, 1, 1]
+    """
```

Comment:
There should be a "see also"  section with references to related functions like I guess `solve_univariate_inequality` or anything else that someone might want to use instead of this for their problem.

_Originally posted by @oscarbenjamin in https://github.com/sympy/sympy/pull/21687#discussion_r703024445_